src/transformStateWithClones.js

```diff
@@ -6,8 +6,39 @@
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
-}
+const transformStateWithClones = (state, actions) => {
+  const results = [Object.assign({}, state)];
+
+  for (const action of actions) {
+    const type = action.type;
+    let newState = Object.assign({}, results[results.length - 1]);
```

Comment:
can be more simplify
```suggestion
    let newState = { ...results };
```